packages/calcite-components/src/components/dialog/dialog.tsx

```diff
@@ -458,11 +458,21 @@ export class Dialog
     this.el.removeEventListener("calciteDialogOpen", this.openEnd);
   };
 
-  private handleScroll = (): void => {
+  private handleScroll = (event: Event): void => {
```

Comment:
To make it clear that we're handling an internal event, can you rename the event handlers to convey this and type this event as `CustomEvent` (similar to `flow-item`)?